include/devicetree.h

```diff
@@ -1258,6 +1258,62 @@
  * @}
  */
 
+/**
+ * @defgroup devicetree-dep-ord Dependency tracking
+ * @ingroup devicetree
+ * @{
+ */
+
+/**
+ * @brief Get a node's dependency ordinal
+ * @param node_id Node identifier
+ * @return the node's dependency ordinal as an unsigned integer literal
+ */
+#define DT_DEP_ORD(node_id) DT_CAT(node_id, _DEP_ORD)
+
+/**
+ * @brief Get a child node's dependency ordinal
+ *
+ * Equivalent to DT_DEP_ORD(DT_CHILD(node_id, child)).
+ *
+ * @param node_id node identifier
+ * @param child lowercase-and-underscores child node name
+ * @return the child node's dependency ordinal
+ * @see DT_CHILD()
+ */
+#define DT_CHILD_DEP_ORD(node_id, child) DT_DEP_ORD(DT_CHILD(node_id, child))
+
+/**
+ * @brief Get the dependency ordinals of all of a node's children
+ *
+ * This expands to a C array initializer containing the dependency
+ * ordinals of the node's children. Each ordinal is terminated by a
+ * comma.
+ *
+ * Example devicetree fragment:
+ *
+ *	n: node {
+ *		child-1 { };
+ *		child-2 { };
+ *		child-3 { };
+ *	};
+ *
+ * Example usage, assuming the child-X nodes have dependency ordinals
+ * c1, c2, and c3 respectively:
+ *
+ *	DT_CHILDREN_DEP_ORD(DT_NODELABEL(n)) // {c1, c2, c3,}
+ *
+ * @param node_id node identifier
+ * @return the dependency ordinals of the node's children as a C array
+ *         initializer
+ */
+#define DT_CHILDREN_DEP_ORDS(node_id) \
```

Comment:
I was unclear about this in part because I was unsure, but I don't think this should be an initializer list, but rather a list, by which I mean I might want to do something like:

    UTIL_LISTIFY(DT_CHILDREN_DEP_ORDS, GEN_HANDLE_SYMBOL);

However, I may be wrong about that.  Early days....

Reply:
OK, I made this a list instead of an array. I left the names unchanged, though. I think "DEP_ORD" is easy to understand as a "dependency ordinal". I think that might be a more practical and readable name since English puts adjectives first.